### PR TITLE
[Img2Img] Fix batch size mismatch prompts vs. init images

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Make sure `init_images` and `prompts` behavior matches.

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py [hunk 1]
```diff
@@ -195,6 +195,7 @@ def __call__(
         """
         if isinstance(prompt, str):
             batch_size = 1
+            prompt = [prompt]
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py:198]: I'm planning on cleaning this up more generally when tackling https://github.com/huggingface/diffusers/issues/551

**Comment by patrickvonplaten** [comment]
@pcuenca @patil-suraj would be great if you could review if you have a minute

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py [hunk 2]
```diff
@@ -284,8 +285,23 @@ def __call__(
...
         init_latents = 0.18215 * init_latents
 
-        # expand init_latents for batch_size
-        init_latents = torch.cat([init_latents] * batch_size * num_images_per_prompt, dim=0)
+        if len(prompt) > init_latents.shape[0] and len(prompt) % init_latents.shape[0] == 0:
+            # expand init_latents for batch_size
+            deprecation_message = (
+                f"You have passed {len(prompt)} text prompts (`prompt`), but only {init_latents.shape[0]} initial"
+                " images (`init_image`). Initial images are now duplicating to match the number of text prompts. Note"
+                " that this behavior is deprecated is will be removed in a version 1.0.0. Please make sure to update"
```
Comment (pcuenca) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py:293]:
```suggestion
                " that this behavior is deprecated and will be removed in a version 1.0.0. Please make sure to update"
```

**Review by pcuenca — APPROVED** [review]
Looks good!

**Pull request merged**